baybe/simulation.py

```diff
@@ -504,23 +495,15 @@ def simulate_experiment(
         try:
             measured = campaign.recommend(batch_size=batch_size)
         except NotEnoughPointsLeftError:
-            # TODO: This except block requires a more elegant solution
-            strategy = campaign.strategy
-            allow_repeated = strategy.allow_repeated_recommendations
-            allow_measured = strategy.allow_recommending_already_measured
-
-            # Sanity check: if the variable was True, the except block should have
-            # been impossible to reach in the first place
-            # TODO: Currently, this is still possible due to bug [15917] though.
-            assert not allow_measured
-
-            measured, _ = campaign.searchspace.discrete.get_candidates(
-                allow_repeated_recommendations=allow_repeated,
-                allow_recommending_already_measured=allow_measured,
+            # TODO: There can be still N < batch_quantity points left in the search
+            #   space. Once the recommender/strategy refactoring is completed,
+            #   find an elegant way to return those.
+            warnings.warn(
+                "The simulation of the campaign ended because there were no more "
```

Comment:
Change to "..because not sufficiently many points were left for recommendation" - this makes it (in my opinion) clearer for the case of `batch_size>1`

Reply:
fine with me